BritneyS/SparkleSpin
# Add player tests

## What you did :question:
This PR adds unit tests in 'ModelTests' for the following operations: 
- Creating a player
- Creating multiple players
- Adding one player to a player list
- Adding multiple players to a player list
- Getting the player list
- Deleting one player
- Deleting a player that has the same name of a previously saved player

## How to test it :microscope:
1. Check out this branch
2. Run the app
3. Push `cmd + U` to run all unit tests
4. Note that all unit tests should pass

## Any background context you want to provide? :question:
- I've removed the original tests I wrote for chore additions for now. Since I restructured how I handled CRUD operations in the `PlayerListViewModel` (formerly known as `PlayerViewModel`) while writing the tests, a similar restructuring will probably be needed for any chore models and view models at a later date.

- I will add tests and functionality for editing a player at a later date.

- While writing the tests and view model methods, I ran into a problem: I wanted a user to be able to delete a player even if that player had the same name as a player from another entry. I explored two solutions: Sets and NSUUID. I learned about the advantages of sets, but I ultimately decided that adding a 'uuid' property to the player model that holds a generated uuid string worked out for my needs. A consequence of this is that for the tests, I could not test for object equality (since a Player instance was now unique by uuid), so I tested for matching name values instead.

## Screenshots (if applicable) :camera:
N/A
